packages/terra-date-input/src/DateInput.jsx

```diff
@@ -0,0 +1,644 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { injectIntl, intlShape } from 'react-intl';
+import Input from 'terra-form-input';
+import KeyCode from 'keycode-js';
+
+import DateInputUtil from './DateInputUtil';
+import styles from './DateInput.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Name of the date input. This name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * Custom input attributes to apply to the day input
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  dayAttributes: PropTypes.object,
+  /**
+   * Whether the date input should be disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Can be used to set the display format. One of `month-day-year` or `day-month-year`.
+   * Overrides default locale display format.
+   */
+  displayFormat: PropTypes.string,
```

Comment:
I don't know how the date format for en-CA would apply for this component. It looks like the [official all-numeric date format](https://en.wikipedia.org/wiki/Date_and_time_notation_in_Canada) for en-CA is YYYY-MM-DD. However, other formats are still used depending on their context. According to [this](https://en.wikipedia.org/wiki/Date_format_by_country), the month still appears before the year when using the long month name. Since this component does not use an all-numeric date format, perhaps, we should can support just these two display formats for now and support other display formats in the future when new requests come in.

Reply:
Talked with @benbcai offline, planning to leave `en-CA` date rendering as month-day-year.